### ISSUE TITLE
Multiple instances and AppModule

### ISSUE THREAD

**Issue by ryanfinken** [tiberiuzuld/angular-busy#2]
Hello,

I use the original cross/angular-busy in my AngularJS 1.6 app, so I was looking for something to work with Angular 5. This appears to be the perfect fit, but I'm having two issues:

1) I can't seem to use multiple busy indicators on the same page. I get the following error when I do: "ERROR TypeError: Cannot read property 'style' of null". The cross version did support this. There is no issue if I only have one [cgBusy] on the page.

2) I'd like to make this available via the AppModule instead of needing to add it to each {feature}Module. Is this possible? 

I'm not doing any custom configuration. Just importing the module and using the [cgBusy] attribute. Any thoughts?

**Comment by tiberiuzuld** [comment]
Hi @ryanfinken ,
1. Will have a look this week.
2. Is an angular issue and I think you only need to add the module to a parent module which imports all feature modules.

**Comment by tiberiuzuld** [comment]
Hi @ryanfinken ,
Issue 1:
Just tested in the demo app and in a separated project with the lib built and couldn't reproduce the issue.
```html
  <li [cgBusy]="promise2">
  </li>
  <li [cgBusy]="promise2">
  </li>
```

**Comment by ryanfinken** [comment]
Hello @tiberiuzuld,

Thank you for your quick response! 

Issue 1: 
I'll take another look and see if I can isolate this further. In your example above, it looks like you are waiting on the same promise in both instances. Does it make a difference for you when you wait on separate promises? Also, does it make any difference if they are observables/subscriptions rather than promises (i am waiting on two separate observables).

Issue 2:
I did try referencing it in the AppModule, which imports all of the feature modules. When I do that, I get this error:

> Uncaught Error: Template parse errors:
> Can't bind to 'cgBusy' since it isn't a known property of 'table'.

That only goes away if I reference it in the actual feature modules instead. It looks like angular2-busy has a similar [issue](https://github.com/devyumao/angular2-busy/issues/9).

Would it be better for tracking if I submit Issue 2 as a separate issue here?

**Comment by tiberiuzuld** [comment]
1. I tested with 2 different promises. It should not matter what you put in.
2. There is nothing I can do. This is how angular works.

**Comment by ryanfinken** [comment]
1. Here is the line of code in CgBusyDirective that is causing the exception (parentNode is null):

> this.renderer.setStyle(this.el.nativeElement.parentNode, 'position', 'relative');

Here is the complete exception:

> ERROR TypeError: Cannot read property 'style' of null
    at DefaultDomRenderer2.setStyle (platform-browser.js:2924)
    at BaseAnimationRenderer.setStyle (animations.js:590)
    at DebugRenderer2.setStyle (core.js:15415)
    at new CgBusyDirective (cgBusy.directive.js:24)
    at createClass (core.js:12480)
    at createDirectiveInstance (core.js:12315)
    at createViewNodes (core.js:13776)
    at createEmbeddedView (core.js:13653)
    at callWithDebugContext (core.js:15090)
    at Object.debugCreateEmbeddedView [as createEmbeddedView] (core.js:14402)

2. I am new to Angular and will have to take your word for it. It is just strange because several other library modules that I use don't seem to have that limitation. For example:

>     // ngx-bootstrap
>     ModalModule.forRoot(),
> 
>     // ngx-webstorage
>     Ng2Webstorage.forRoot({
>       separator: '.',
>       caseSensitive: true
>     }),
> 
>     // ngx-toastr
>     ToastrModule.forRoot({
>       timeOut: 10000,
>       positionClass: 'toast-bottom-right',
>       preventDuplicates: true
>     }),

These are all imported one time in AppModule and have no need to be imported in the feature modules. Maybe implementing the forRoot is the missing link?

**Comment by ryanfinken** [comment]
Figured out the first issue! The element I had it on was being affected by an ngIf. Oops! I can see how that would cause issues. I'm sorry for the trouble.

**Comment by tiberiuzuld** [comment]
Glad to hear that the issue 1 was resolved.
About issue 2:
All your examples are using a service not a directive/component.
`forRoot` is used for something else:
https://stackoverflow.com/questions/39653072/how-to-use-forroot-within-feature-modules-hierarchy

This library uses a directive.
The angular documentation clearly states at `exports` property: 

> Specifies a list of directives/pipes/modules that can be used within the template of any component that is part of an Angular module that imports this Angular module.

https://angular.io/api/core/NgModule#example-1


There is nothing I can do.

**Comment by ryanfinken** [comment]
OK, that makes more sense to me now. Thank you for your responsiveness and for the explanation!
